app/controllers/applicationController.js

```diff
@@ -74,7 +96,31 @@ updateApplication = async (req, res) => {
 };
 
 deleteApplication = async (req, res) => {
+    const id = req.params.id;
+
+    const application = await Application.findById(id).catch(() => {
+        res.status(404).json({
+            message: `Application ${id} not found`,
+        });
+    });
+    if (!application) {
+        return res.status(201).json({
+            message: `Application does not exist`,
+        });
+    }
 
+    Application.deleteOne({ _id: application._id })
+        .then(() => {
+            return res.status(201).json({
+                message: `Application deleted`,
+            });
+        })
+        .catch(error => {
+            return res.status(400).json({
+                error,
+                message: "There was an error deleting your application"
+            });
+        })
 };
```

Comment:
Trying to double set the response code when an application isn't found has a 50% chance of crashing my local server. I think we'd be okay sticking with the first message!

```suggestion
    if (application) {
        Application.deleteOne({ _id: application._id })
            .then(() => {
                return res.status(201).json({
                    message: `Application deleted`,
                });
            })
            .catch(error => {
                return res.status(400).json({
                    error,
                    message: "There was an error deleting your application"
                });
            })
    }
};
```

Reply:
I found that if I didn't check if the application existed before trying to delete my server would just spin before timing out because it would be trying to delete something that doesn't exist.

Reply:
I removed the section where I could have been double setting the response. But I left the application check. When trying to find the application it will not end up in the catch block unless the findById() fails. When findById() simply does not find something that is _not_ a failure so the catch block is never hit.